### ISSUE TITLE
ROP: ReduceOnPlateau Series

### ISSUE THREAD

**Issue by yzbx** [ISCAS007/torchseg#33]
# code
```
10674* python test/pspnet_test.py --backbone_pretrained=True --use_momentum=True --learning_rate=1e-3 --use_lr_mult=True --midnet_scale=5 --upsample_type=bilinear --upsample_layer=4 --focal_loss_gamma=2.0 --backbone_name=resnet50 --use_dropout=False --use_bias=True --momentum=0.01 --use_class_weight=False --optimizer=adam --changed_lr_mult=2.0 --new_lr_mult=5.0 --note=fl_poly_rop --n_epoch=250 --save_model=True --batch_size=4 --scheduler=poly_rop

10675* python test/pspnet_test.py --backbone_pretrained=True --use_momentum=True --learning_rate=1e-2 --use_lr_mult=True --midnet_scale=10 --upsample_type=bilinear --upsample_layer=4 --focal_loss_gamma=2.0 --backbone_name=resnet50 --use_dropout=False --use_bias=True --momentum=0.01 --use_class_weight=True --class_weight_alpha=0.1 --optimizer=sgd --changed_lr_mult=2.0 --new_lr_mult=5.0 --note=flcw_poly_rop --n_epoch=250 --save_model=True --batch_size=4 --scheduler=poly_rop

10676* python test/pspnet_test.py --backbone_pretrained=True --use_momentum=True --learning_rate=1e-2 --use_lr_mult=True --midnet_scale=5 --upsample_type=bilinear --upsample_layer=4 --focal_loss_gamma=2.0 --backbone_name=resnet50 --use_dropout=False --use_bias=True --momentum=0.01 --use_class_weight=False --optimizer=sgd --changed_lr_mult=2.0 --new_lr_mult=5.0 --note=fl_poly_rop --n_epoch=250 --save_model=True --batch_size=4 --scheduler=poly_rop

10677* python test/pspnet_test.py --backbone_pretrained=True --use_momentum=True --learning_rate=1e-2 --use_lr_mult=True --midnet_scale=5 --upsample_type=bilinear --upsample_layer=5 --focal_loss_gamma=2.0 --backbone_name=resnet50 --use_dropout=False --use_bias=True --momentum=0.01 --use_class_weight=True --class_weight_alpha=0.1 --optimizer=sgd --changed_lr_mult=2.0--new_lr_mult=5.0 --note=flcw_poly_rop_ft --n_epoch=250 --save_model=True --checkpoint_path=/home/yzbx/tmp/logs/pytorch/pspnet/cityscapes/fl_cw_alpha7/2018-11-09___13-57-00/model-last-249.pkl --batch_size=4 --scheduler=poly_rop
```

**Comment by yzbx** [comment]
# results

|    |   train/iou |   val/iou | note           |   upsample_layer |   midnet_scale |   n_epoch |
|---:|------------:|----------:|:---------------|-----------------:|---------------:|----------:|
|  2 |    0.518915 |  0.436921 | fl_plateau_ul4 |                4 |              5 |       ~~250~~(80) |
|  1 |    0.568463 |  0.443785 | fl_plateau_ul4 |                4 |              5 |       ~~250~~(80) |
|  3 |    0.504173 |  0.454455 | plateau        |                4 |             10 |       ~~250~~(80) |
|  0 |    0.755878 |  0.515256 | fl_plateau_ft  |                5 |              5 |       ~~250~~(80) |


|    |   train/iou |   val/iou | note             |   upsample_layer |   midnet_scale |   n_epoch |
|---:|------------:|----------:|:-----------------|-----------------:|---------------:|----------:|
|  2 |    0.410586 |  0.392286 | flcw_poly_rop    |                4 |             10 |       250 |
|  3 |    0.553842 |  0.431579 | flcw_poly_rop_ft |                5 |              5 |       250 |
|  1 |    0.669754 |  0.463747 | fl_poly_rop      |                4 |              5 |       250 |
|  0 |    0.563162 |  0.463881 | fl_poly_rop      |                4 |              5 |       250 |


|    |   train/iou |   val/iou | note     |   upsample_layer |   midnet_scale |   n_epoch |
|---:|------------:|----------:|:---------|-----------------:|---------------:|----------:|
|  0 |    0.600995 |  0.517956 | flcw_pop |                4 |             10 |       250 |

**Comment by yzbx** [comment]
# problems
- too slow on val miou --> step on val loss
- faster converge --> more patience
